### ISSUE TITLE
`error` at the empty type

### ISSUE THREAD

**Issue by weaversa** [GaloisInc/cryptol#1160]
```
Cryptol> error "This is an error" : [0]
0x0
```

This is kind of strange. I have something that works like:

```
f a b = if a then error "oh no!" else b
```
And I want the error to occur when `a == True`, even when `b` has type `[0]'.

Though, would we need to be careful to keep `undefined` working as it currently does?

**Comment by robdockins** [comment]
This is a remnant of the old error semantics where errors and nontermination were only supposed to happen at base types like `Bit` and `Integer`.  I "supposed to" because we never fully got it to work right.  We have been working for some time to change to a semantics where errors and nontermination are treated uniformly at all types, which I think is a lot less surprising.  I believe this particular problem will be fixed once I merge #1136, which fixes all the semantic issues in this area (that I know of).

**Comment by robdockins** [comment]
Fixed via #1136
